zonemaster/zonemaster-engine
# Remove deprecated features planned for v2023.2

## Purpose

This PR removes deprecated features that were planned for v2023.2 See the **Changes** section below.

## Context

Module `Zonemaster::Engine::Net::IP` was deprecated by https://github.com/zonemaster/zonemaster-engine/pull/1202
Profile options `dnssec` and `edns_size` were deprecated by https://github.com/zonemaster/zonemaster-engine/pull/1147

## Changes

- Remove module `Zonemaster::Engine::Net::IP`
- Remove profile options `dnssec` and `edns_size`
- Update unitary tests

## How to test this PR

Tests are updated and should pass.

**Manual testing:**

```
$ git log -1 --oneline
910e9c5e (HEAD -> clean-deprecated, origin/clean-deprecated) Remove deprecated features planned for v2023.2
```

1) Check that `Zonemaster::Engine::Net::IP` is not available anymore

```
$ perl -MZonemaster::Engine -E 'use Zonemaster::Engine::Net::IP'
Can't locate Zonemaster/Engine/Net/IP.pm in @INC (you may need to install the Zonemaster::Engine::Net::IP module)
```

2) Check that setting profile option `dnssec` or `edns_size` in a custom profile return an error

```
$ cat share/custom.json
{
    "resolver" : {
        "defaults" : {
           "debug" : false,
           "dnssec" : false,
           "edns_size" : 0
        }
    },
    "test_cases": [
        "address01"
    ]
}

$ zonemaster-cli zonemaster.net --profile share/custom.json
Loading profile from share/custom.json.
Unknown property 'resolver.defaults.edns_size' at /home/tgreen/zonemaster/zonemaster-engine/lib/Zonemaster/Engine/Profile.pm line 311.
```
